Source/WebCore/dom/Document.cpp

```diff
@@ -6904,6 +6904,11 @@ void Document::serviceRequestAnimationFrameCallbacks()
         m_scriptedAnimationController->serviceRequestAnimationFrameCallbacks(domWindow()->frozenNowTimestamp());
 }
 
+void Document::serviceCaretAnimation()
+{
+    m_selection->currentCaretAnimation()->serviceCaretAnimation(domWindow()->frozenNowTimestamp());
```

Comment:
Can we be sure that all these pointers are non-null?

Reply:
Yes; `m_selection` is a `UniqueRef`, and currentCaretAnimation is always initialized in the FrameSelection constructor

Reply:
You could use `selection()` instead of `m_selection` to make it clear it's a reference.

I don't think `domWindow()` is guaranteed to be non-null though.